types/signature_cache.go

```diff
@@ -0,0 +1,36 @@
+package types
+
+// The value type for the verified signature cache.
+type SignatureCacheValue struct {
+	ValidatorPubKeyBytes []byte
+	VoteSignBytes        []byte
+}
+
+type SignatureCache interface {
+	Add(key string, value SignatureCacheValue)
+	Get(key string) (SignatureCacheValue, bool)
+	Len() int
+}
+
+type signatureCache struct {
+	cache map[string]SignatureCacheValue
+}
+
+func NewSignatureCache() SignatureCache {
```

Comment:
I don't see how this is better than the original `map[string]SignatureCacheValue`. @cason is this what you had in mind?

Reply:
Yes, so we can implement this abstraction in the way we prefer.